functions/src/index.ts

```diff
@@ -172,8 +172,19 @@ exports.calculateAqi = functions.pubsub
       };
 
       if (currentSensorData.isActive) {
+        // Data used to calculate hourly averages
+        const pm25BufferStatus: bufferStatus =
+          sensorDocData.pm25BufferStatus ?? bufferStatus.DoesNotExist;
+        const pm25BufferIndex: number = sensorDocData.pm25BufferIndex ?? 0;
+        const pm25Buffer: Array<Pm25BufferElement> =
+          sensorDocData.pm25Buffer ?? [];
+
         // Get current sensor readings
-        const hourlyAverages = await getHourlyAverages(sensorDoc.id);
+        const hourlyAverages = getHourlyAverages(
```

Comment:
nit: just for readability, can you add the type, `hourlyAverages: SensorReading[]`